cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -380,26 +355,14 @@ ObjectId resolveHead() throws IOException {
     return git.getRepository().resolve(Constants.HEAD);
   }
 
-  /**
-   * Pops the stash with the given rev commit.
-   */
-  private void popStash(@Nullable RevCommit stashID) throws GitAPIException {
-    if (stashID == null) {
-      LOG.debug("Pop stash called with null stashID, nothing will be popped.");
-      return;
-    }
-    // pop the stash.
-    git.stashApply().setStashRef(stashID.getName()).setRestoreUntracked(true).call();
-    Collection<RevCommit> stashes = git.stashList().call();
-    int stashIndex = 0;
-    for (RevCommit stash : stashes) {
-      if (stash.equals(stashID)) {
-        git.stashDrop().setStashRef(stashIndex).call();
-        return;
-      }
-      stashIndex++;
+  @Nullable
+  private String getFileHash(Path relativePath, RevCommit commit) throws IOException {
+    Path absolutePath = getRepositoryRoot().resolve(relativePath);
+    // Find the node representing the exact file path in the tree.
+    TreeWalk walk = TreeWalk.forPath(git.getRepository(), relativePath.toString(), commit.getTree());
+    if (walk == null) {
+      return null;
     }
-    LOG.warn("Couldn't find Git stash with ID {} to drop. Ignoring missing stash.", stashID.getName());
+    return walk.getObjectId(0).getName();
```

Comment:
Do we need null check for getObjectid here?

Reply:
I checked the code for `TreeWalk`. The last argument to `TreeWalk#forPath()` can be a variadic list of RevTrees. `getObjectID(0)` returns the object ID in the 0th RevTree passed in `forPath`. 
If the Object isn't present in the 0th tree, it will return `ObjectId.zeroId()` which is a static constant in `ObjectId` class: 
```
private static final ObjectId ZEROID = new ObjectId(0, 0, 0, 0, 0)
```
However, this case never seems to happen as the TreeWalk never goes down a path which doesn't exist in the given RevTrees. So we get a `null` as the result of `TreeWalk#forPath()` when the path doesn't exist. In other cases we should get a non-null result from `TreeWalk#getObjectId(0)`